WalletWasabi.Tests/PeriodicRunnerTests/SerialPeriodicRunnerTests.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Bases;
+using Xunit;
+
+namespace WalletWasabi.Tests.PeriodicRunnerTests
+{
+	/// <summary>
+	/// The tests in this collection are time-sensitive, therefore this test collection is run in a special way:
+	/// Parallel-capable test collections will be run first (in parallel), followed by parallel-disabled test collections (run sequentially) like this one.
+	/// </summary>
+	/// <seealso href="https://xunit.net/docs/running-tests-in-parallel.html#parallelism-in-test-frameworks"/>
+	[Collection("Serial unit tests collection")]
+	public class SerialPeriodicRunnerTests
+	{
+		internal class TestRunner : PeriodicRunner
+		{
+			internal TestRunner(TimeSpan period, TimeSpan maxNextRoundWaitTime) : base(period)
+			{
+				MaxNextRoundWaitTime = maxNextRoundWaitTime;
+			}
+
+			public SemaphoreSlim Semaphore { get; private set; } = new SemaphoreSlim(initialCount: 0, maxCount: 1);
+
+			public int RoundCounter { get; private set; }
+			private TimeSpan MaxNextRoundWaitTime { get; }
+
+			protected override async Task ActionAsync(CancellationToken cancel)
+			{
+				// Add some delay to simulate work.
+				await Task.Delay(50, cancel).ConfigureAwait(false);
+				RoundCounter++;
+
+				Semaphore.Release();
+			}
+
+			public async Task<bool> WaitForNextRoundAsync()
+			{
+				return await Semaphore.WaitAsync(MaxNextRoundWaitTime);
+			}
+		}
+
+		[Fact]
+		public async Task PeriodicRunnerTestsAsync()
+		{
+			const double Scaler = 5.0;
+			TimeSpan leniencyThreshold = Scaler * TimeSpan.FromSeconds(0.5);
+			TimeSpan period = Scaler * TimeSpan.FromSeconds(1);
+
+			using var runner = new TestRunner(period: period, maxNextRoundWaitTime: TimeSpan.FromSeconds(Scaler * 2));
+			using CancellationTokenSource cts = new CancellationTokenSource();
```

Comment:
Be consistent. If in the previous line you used `using var runner` do the same here and put `using var cts`